controller/test/frame_detector/frame_detector_test.cpp

```diff
@@ -0,0 +1,144 @@
+#include "frame_detector.h"
+#include "proto_traits.h"
+#include "soft_rx_buffer.h"
+#include "gtest/gtest.h"
+
+constexpr uint32_t LEN = 20;
+
+constexpr FrameDetector<SoftRxBuffer<LEN>, LEN>::State STATE_LOST =
+    FrameDetector<SoftRxBuffer<LEN>, LEN>::State::LOST;
+constexpr FrameDetector<SoftRxBuffer<LEN>, LEN>::State STATE_WAIT_START =
+    FrameDetector<SoftRxBuffer<LEN>, LEN>::State::WAIT_START;
+constexpr FrameDetector<SoftRxBuffer<LEN>, LEN>::State STATE_RX_FRAME =
+    FrameDetector<SoftRxBuffer<LEN>, LEN>::State::RX_FRAME;
+
+constexpr uint8_t MARK = static_cast<uint8_t>('.');
+
+SoftRxBuffer<LEN> rx_buf('.');
+
+TEST(FrameDetector, TrivialRx) {
```

Comment:
These tests need some more explanation of what situations they are simulating and why the assertions are what they are.

Reply:
Please address this

Reply:
Well, it was a trivial test, now it does not make sense as we have fakeRx. Test removed.